server/admin.go

```diff
@@ -113,32 +113,24 @@ func (s *adminServer) handleDebug(w http.ResponseWriter, r *http.Request) {
 
 // handleRESTAction handles RESTful admin actions.
 func (s *adminServer) handleRESTAction(handler actionHandler, w http.ResponseWriter, r *http.Request, prefix string) {
+	path, err := url.QueryUnescape(strings.TrimPrefix(r.URL.Path, prefix))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
Note that a bad path now supersedes a bad method. Allowed for removing duplicate calls though.